### PR TITLE
Add support for creating non namespaced resources

### PR DESCRIPTION
**Pull request by michaelhtm**
Description of changes:
This change will ensure we find out whether a resource 
is namespaced or not before attempting to create it

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: internal/resourcegroup/builder.go [hunk 1]
```diff
@@ -91,6 +100,17 @@ func (b *GraphBuilder) NewResourceGroup(rg *v1alpha1.ResourceGroup) (*ResourceGr
 	// 3. Extract CEL expressions from the schema.
 	// 4. Build the resource object.
 
+	namespacedResources := map[k8sSchema.GroupVersionKind]bool{}
+
+	apiResourceList, err := b.discoveryClient.ServerPreferredNamespacedResources()
```
Comment (a-hilaly) [inline internal/resourcegroup/builder.go:100]: errcheck 🚓

File: internal/resourcegroup/builder.go [hunk 2]
```diff
@@ -50,13 +52,20 @@ func NewResourceGroupBuilder(
...
 type GraphBuilder struct {
 	schemaResolver   resolver.SchemaResolver
 	resourceEmulator *emulator.Emulator
+	discoveryClient  *discovery.DiscoveryClient
+}
+
+type GroupVersionKind struct {
+	apiGroupVersion string
+	kind            string
 }
```
Comment (a-hilaly) [inline internal/resourcegroup/builder.go]: Can we just use the official GVK struct?

File: internal/controller/instance/controller_reconcile.go [hunk 3]
```diff
@@ -182,8 +182,16 @@ func (igr *InstanceGraphReconciler) reconcileResource(ctx context.Context, resou
...
-	rc := igr.client.Resource(gvr).Namespace(namespace)
+
+	var rc dynamic.ResourceInterface
+	var namespace string
+	if igr.rg.Resources[resourceID].Namespaced {
+		namespace = igr.getResourceNamespace(resourceID)
+		rc = igr.client.Resource(gvr).Namespace(namespace)
+	} else {
+		rc = igr.client.Resource(gvr)
+		namespace = "non-namespaced"
```
Comment (a-hilaly) [inline internal/controller/instance/controller_reconcile.go]: I'm thinking.. should we keep this empty? `"non-namepaced"` is a correct namespace..

File: internal/resourcegroup/builder.go (hunk 2 above)
Comment (michaelhtm) [reply internal/resourcegroup/builder.go]: forgot to remove this

File: internal/controller/instance/controller_reconcile.go (hunk 3 above)
Comment (michaelhtm) [reply internal/controller/instance/controller_reconcile.go]: ohh I was unsure what to put here lol

**Pull request merged**